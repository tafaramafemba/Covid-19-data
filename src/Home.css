@import url('https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,100;0,300;0,400;1,100;1,300;1,400&family=Montserrat:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;1,100;1,200;1,300;1,400;1,500;1,600&family=Roboto+Slab:wght@900&display=swap');

.home-ul {
  display: grid;
  grid-template-columns: auto auto;
  background-color: #4369a5;
  margin-top: 0;
  padding-left: 0;
  justify-content: center;
  padding-bottom: 30px;
  margin-bottom: 0;
}

.country-items {
  display: flex;
  flex-direction: column;
}

.virus {
  width: 75px;
  height: 75px;
}

.country-li {
  font-family: 'Lato', sans-serif;
  color: #fff;
  padding-top: 50px;
  padding-bottom: 50px;
  width: 96%;
  border-left: 1px solid #fff;
  border-bottom: 1px solid #fff;
  margin-right: 43px;
}

a:link {
  list-style: none;
  text-decoration: none;
}

.img {
  width: 100px;
  height: 100px;
}

.title-container {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  gap: 50px;
  padding-top: 15px;
}

.south-text {
  font-family: 'Lato', sans-serif;
  font-size: 30px;
  color: #4369b2;
}

.stats {
  background-color: #304b7e;
  width: 100%;
  font-family: 'Lato', sans-serif;
  padding-top: 5px;
  padding-bottom: 5px;
  font-weight: bold;
  color: #fff;
  text-align: center;
}
