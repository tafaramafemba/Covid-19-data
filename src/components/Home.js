import { useEffect } from 'react';
import { useDispatch, useSelector} from 'react-redux';
import { showCountries } from '../redux/home/home';
import { useNavigate } from 'react-router-dom';
import '../Home.css';

const Home = () => {

  const southern = [
    'Angola',
    'Botswana',
    'Lesotho',
    'Mozambique',
    'Namibia',
    'South Africa',
    'Swaziland',
    'Zambia',
    'Zimbabwe',
  ];

  const state = useSelector((state) => state.country.filter((country) => southern.includes(country.name)));
  const dispatch = useDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(showCountries());
  }, []);

  const handleRedirect = (country) => {
    navigate(`/${country}`);
  };

  return (
    <div>
      <div className='title-container'>
        <div className='map-backgrnd'>
        <img src='map.png' alt='map' className='img'></img>
        </div>
        <div>
        <span className='south-text'>Southern Africa</span>
        </div>
      </div>
      <div className='stats'>
      <span className='stats'>Stats by country</span>
      </div>
      <ul className='home-ul'>
        {state.map((country) => (
          <li className='country-li' key={country.id} onClick={() => handleRedirect(country.name)}>
              <img className='virus' src='virus.png'></img>
              <div className='country-items'>
              <h3>{country.name}</h3>
              <span>New cases: {country.today_confirmed}</span>
              </div>
              
          </li>
        ))}
      </ul>
    </div>
  )
}

export default Home;