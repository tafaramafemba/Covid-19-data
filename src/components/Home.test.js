import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Home from './Home';


describe(' Home Component Rendering ', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});