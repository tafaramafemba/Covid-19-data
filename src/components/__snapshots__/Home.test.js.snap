// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Home Component Rendering  renders correctly 1`] = `
<div>
  <div
    className="title-container"
  >
    <div
      className="map-backgrnd"
    >
      <img
        alt="map"
        className="img"
        src="map.png"
      />
    </div>
    <div>
      <span
        className="south-text"
      >
        Southern Africa
      </span>
    </div>
  </div>
  <div
    className="stats"
  >
    <span
      className="stats"
    >
      Stats by country
    </span>
  </div>
  <ul
    className="home-ul"
  />
</div>
`;
